From 4f5f66750a809308c407924ad8abc146838518b9 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 17:33:09 -0400
Subject: [infiniband] ipath: Translate legacy diagpkt into newer extended diagpkt

Message-id: <13e7fc479310b9796e2232e68dcce90782c0622a.1411579814.git.dledford@redhat.com>
Patchwork-id: 96702
O-Subject: [Patch RHEL7 16/19] IB/ipath: Translate legacy diagpkt into newer extended diagpkt
Bugzilla: 1085961
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Dennis Dalessandro <dennis.dalessandro@intel.com>

This patch addresses an issue where the legacy diagpacket is sent in
from the user, but the driver operates on only the extended
diagpkt. This patch specifically initializes the extended diagpkt
based on the legacy packet.

Cc: <stable@vger.kernel.org>
Reported-by: Rickard Strandqvist <rickard_strandqvist@spectrumdigital.se>
Reviewed-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
Signed-off-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 7e6d3e5c70f13874fb06e6b67696ed90ce79bd48)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ipath/ipath_diag.c b/drivers/infiniband/hw/ipath/ipath_diag.c
index e2f9a51..45802e9 100644
--- a/drivers/infiniband/hw/ipath/ipath_diag.c
+++ b/drivers/infiniband/hw/ipath/ipath_diag.c
@@ -346,6 +346,10 @@ static ssize_t ipath_diagpkt_write(struct file *fp,
    ret = -EFAULT;
    goto bail;
   }
+  dp.len = odp.len;
+  dp.unit = odp.unit;
+  dp.data = odp.data;
+  dp.pbc_wd = 0;
  } else {
   ret = -EINVAL;
   goto bail;
-- 
1.7.1